libgalois/src/GraphTopology.cpp

```diff
@@ -435,6 +435,192 @@ katana::EdgeTypeAwareTopology::MakeFrom(
       pg, edge_type_index, e_topo, std::move(per_type_adj_indices)});
 }
 
+std::unique_ptr<katana::ProjectedShuffleTopology>
+katana::ProjectedShuffleTopology::Make(
+    const katana::PropertyGraph* pg, const std::vector<std::string>& node_types,
```

Comment:
This appears to be shuffling based on the type specifically. But the name just says that this is a shuffled topology. If the type is supposed to always be used for type sorting, this should be called `TypeSortedTopology` or something. If other shuffles are allow (like degree order or whatever), we need to provide ways to create views for any arbitrary shuffle and make sure we can cache them all safely.

Reply:
this class is baically just projecting the node and edge types, so may be `ProjectedTypeToplogy` is a better name.

Reply:
Yes, `Shuffle` is not appropriate here.

Reply:
so I will make it `ProjectedTypeTopology` then